source/tests/system/nirfmxbt_driver_api_tests.cpp

```diff
@@ -211,6 +211,32 @@ TEST_F(NiRFmxBTDriverApiTests, TxpBasicFromExample_DataLooksReasonable)
   EXPECT_GT(fetched_powers_response.peak_to_average_power_ratio_maximum(), 0.0);
 }
 
+TEST_F(NiRFmxBTDriverApiTests, SetAndGetAttributeInt32_Succeeds)
+{
+  auto session = init_session(stub(), kPxi5663e);
+  EXPECT_SUCCESS(session, client::cfg_frequency_reference(stub(), session, "", FrequencyReferenceSource::FREQUENCY_REFERENCE_SOURCE_ONBOARD_CLOCK, 10e6));
```

Comment:
Since this setup is the same routine as in `AcpBasicFromExample_DataLooksReasonable()`, I want to move the routine to a helper function. However, I have been running into several issues with doing so. Even on successful builds, the test fails with this error when I use a helper function:

`-380413: Invalid Acquisition State Transition.\r\n\nniRFInstr SA HW UTIL Transition Hardware State.vi <append>\nRFmx driver cannot transition to the  Configuring  state from the Acquiring state. The Acquisition must complete or Aborted.\n"`

I think there's probably a way to do this, but after talking to Alex more I want to submit what I have, get VSCode and ninja set up so that I can build more efficiently, and then I'll create a separate task to come back and single-source this setup routine.

Reply:
I'd expect this to look more like:

```cpp

  constexpr auto ATTR_ID = NiRFmxBTAttribute::NIRFMXBT_ATTRIBUTE_TXP_WHATEVER;
  constexpr auto ATTR_VAL = WHATEVER;
  auto session = init_session(stub(), kPxi5663e);
  EXPECT_SUCCESS(session, client::select_measurements(stub(), session, "", MeasurementTypes::MEASUREMENT_TYPES_TXP, Boolean::BOOLEAN_TRUE));
  EXPECT_SUCCESS(
      session,
      client::set_attribute_i32(stub(), session, "", ATTR_ID, ATTR_VAL));

  EXPECT_SUCCESS(session, client::initiate(stub(), session, "", ""));

  auto get_response = client::get_attribute_i32(stub(), session, "", ATTR_ID);

  EXPECT_SUCCESS(session, get_response);
  EXPECT_EQ(ATTR_VAL, get_response.attr_val());

```

1. Remove extra configuration.
2. Call set_attribute before initiate and get_attribute after
3. Get/Set an attribute that matches the measurement type (TXP not ACP)
4. (I don't really care about the constants declared at the top; that's just to get the WHATEVER's in one place)

Does something like that work?

Reply:
Ugh, it took me way too long to realize that I was mixing up the measurement types. I think that was the root of my difficulty with this change, and originated from a copy/paste error on my part.

Yeah, I tried implementing your suggested approach, and the test passes!